lib/crawler/api/config.rb

```diff
@@ -292,6 +296,30 @@ def validate_domain!(domain)
         raise ArgumentError, "Domain #{domain.inspect} cannot have a path" unless url.path == ''
       end
 
+      def configure_crawl_rules!
+        @crawl_rules = domains.each_with_object({}) do |domain, crawl_rules|
+          url = domain[:url]
+          if domain[:crawl_rules].nil?
+            crawl_rules[url] = {}
+            next
+          end
+
+          raise ArgumentError, "Crawl rules for #{url} is not an array" unless domain[:crawl_rules].is_a?(Array)
```

Comment:
```suggestion
          if domain[:crawl_rules].nil?
            crawl_rules[url] = {}
            domain[:crawl_rules] = []
            next
          end

          raise ArgumentError, "Crawl rules for #{url} is not an array" unless domain[:crawl_rules].is_a?(Array)
```

Otherwise, anything that matches that `if` will end up raising, right?

Reply:
We're iterating over `domains` which is the user's configuration taken from the yaml file. It's okay if one of these values is `nil`, it just means the user didn't add a crawl rule for that domain.

I think the variable naming here is causing confusion (we have `crawl_rules` and `domains[:crawl_rules]` which are different things). I'll rename some vars.

Reply:
🤦 missed the `next`